client/odie/message/index.tsx

```diff
@@ -73,15 +107,57 @@ const ChatMessage = ( { message, isLast, messageEndRef }: ChatMessageProps ) =>
 				</div>
 			) }
 			{ message.type === 'introduction' && (
-				<>
-					<div className="odie-introduction-message-content">
-						<img src={ WapuuAvatar } alt="Wapuu profile" className="odie-chatbox-message-avatar" />
-						<div className="odie-chatbox-introduction-message">{ message.content }</div>
-					</div>
-				</>
+				<div className="odie-introduction-message-content">
+					<img src={ WapuuAvatar } alt="Wapuu profile" className="odie-chatbox-message-avatar" />
+					<div className="odie-chatbox-introduction-message">{ message.content }</div>
+				</div>
 			) }
 		</div>
 	);
+
+	useEffect( () => {
+		const backdropCopy = backdropRef.current;
+		if ( isFullscreen ) {
+			backdropRef.current?.addEventListener( 'click', handleBackdropClick );
+		}
+		return () => {
+			backdropCopy?.removeEventListener( 'click', handleBackdropClick );
+		};
+	}, [ isFullscreen ] );
+
+	const fullscreenContent = (
+		<div className="odie-fullscreen" ref={ backdropRef } onClick={ handleBackdropClick }>
```

Comment:
Can't we use `onClick` here and get rid of the ref? Refs are a constant source of bugs.